echo/index.js

```diff
@@ -181,13 +181,20 @@ function Client() {
     debug('request options', requestOptions);
 
     request.post(requestOptions, function onResp(err, response, body) {
-      var statusCode = response && response.statusCode
-        ? response.statusCode
-        : 0;
-
-      if (err || parseInt(statusCode / 100) !== 2) {
-        error('[echoClient] addEvents error', { err: err, body: body, statusCode: statusCode });
-        callback(err || 'error response status code: ' + statusCode);
+      if (err) {
+        var errorResponse = {
+          statusCode: 0,
+          message: err.message 
+        };
+        error('[echoClient] addEvents error', { error: errorResponse.message, statusCode: errorResponse.statusCode, body: body });
+        callback(errorResponse);
```

Comment:
can you change this to `return callback(errorResponse);` we've had issues in the past where not explicitly returning can result in the situation where lines after this are executed if they are present. Do the same for the others in this p/r

Reply:
I have seen conflicting statements about this, from what I understand the preferred method amongst most platform team members is:
```javascript
callback(errorResponse);
return;
```